src/components/AnalyticsSearch/index.tsx

```diff
@@ -283,15 +286,29 @@ const AnalyticsSearch: React.FC = () => {
         <div ref={wrapperRef} className='searchWidgetContent'>
           <h6>{t('pairs')}</h6>
           {filteredPairs.slice(0, pairsShown).map((val, ind) => {
-            const currency0 = new Token(
+            const currency0 = getTokenFromAddress(
+              val.token0.id,
               MATIC_CHAIN,
-              getAddress(val.token0.id),
-              val.token0.decimals,
+              tokenMap,
+              [
+                new Token(
+                  MATIC_CHAIN,
+                  getAddress(val.token0.id),
+                  val.token0.decimals,
+                ),
+              ],
             );
-            const currency1 = new Token(
+            const currency1 = getTokenFromAddress(
+              val.token1.id,
               MATIC_CHAIN,
-              getAddress(val.token1.id),
-              val.token1.decimals,
+              tokenMap,
+              [
+                new Token(
+                  MATIC_CHAIN,
+                  getAddress(val.token1.id),
```

Comment:
Why do we need to call get address if you just grabbed the token from the address